frontend/src/components/CourseList/CourseListSlice.ts

```diff
@@ -0,0 +1,84 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { AppThunk} from "../../app/store";
+import { CourseCreateObject } from "../Course/CourseDetailsSlice";
+import CoursesService, { CourseListElementDto } from "../../api/courses.service";
+import * as ActiveCourse  from '../../app/ActiveCourse';
+
+
+export interface CourseListElementModel extends CourseCreateObject {
+  _id: string;
+}
+
+interface CourseListState {
+  courseList: CourseListElementModel[];
+  activeCourse?: CourseListElementModel;
+}
+
+const initialState: CourseListState = {
+  courseList: [],
+  activeCourse: ActiveCourse.getActiveCourse()
+};
+
+export const fetchCoursesAsync = (): AppThunk => async (dispatch) => {
+  const courses = await fetchCoursesAndSort();
+  dispatch(setCourses(courses));
+};
+
+export const fetchCoursesAndSort = async () => {
+  const courseService = new CoursesService();
+  const response = await courseService.fetchCourses();
+  const coursesDto: CourseListElementDto[] = response.data;
+  const courses: CourseListElementModel[] = coursesDto.map((courseDto) => {
+    const courseListElement: CourseListElementModel = {
+      _id: courseDto._id,
+      name: courseDto.name,
+      description: courseDto.description,
+      endDate: new Date(Date.parse(courseDto.endDate)), 
+      startDate: new Date(Date.parse(courseDto.startDate)),
+    };
+    return courseListElement;
+  });
+  courses.sort(function (courseListElement1, courseListElement2) {
+    var courseListElementStartDate1 = courseListElement1.startDate.getTime();
+    var courseListElementStartDate2 = courseListElement2.startDate.getTime();
```

Comment:
It would be better to use `const` instead of `var`.

Reply:
Done